Pokemon Unity/Assets/Scripts/Save/GlobalSaveManager.cs

```diff
@@ -0,0 +1,275 @@
+                            else
+                            {
+                                //Try to convert the created file into the current version
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            Debug.Log(e.ToString());
+                        }
+                    }
+                }
+            }
+            catch (IOException e)
+            {
+                throw e;
+            }
+        }
+
+        saveFiles = saveFiles.OrderBy(x => x.TimeCreated).ToList();
+        if (Amount == 0 || saveFiles.Count < Amount)
+        {
+            return saveFiles;
+        }
+        else
+        {
+            return saveFiles.Take(Amount).ToList();
+        }
+    }
+
+    [System.Serializable]
+    private class CustomSaveData
+    {
+        public string BuildVersion = GlobalSaveManager.BuildVersion;
+        public string SaveName = string.Empty;
+
+        public DateTime TimeCreated;
+
+        //Player
+        public SerializableVector3 PlayerPosition;
+        public int PlayerDirection;
+        //Follower
+        public SerializableVector3 FollowerPosition;
+        public int FollowerDirection;
+
+        public int ActiveScene;
+
+        /// <summary>
+        /// When the Player blacks out they need to return to the last visited Pokemon Center
+        /// </summary>
+        private int pCenterScene;
+        private SerializableVector3 pCenterPosition;
+        private int pCenterDirection;
+
+        private SerializableVector3 pCenterFposition;
+        private int pCenterFdirection;
+
+        public Pokemon[][] Party;
+        public Bag PlayerBag;
+
+        public List<CustomSaveEvent> SaveData;
+
+        public void AddPokemonCenter(int scene, SerializableVector3 position, int direction, SerializableVector3 fPosition, int fDirection)
+        {
+            pCenterScene = scene;
+
+            pCenterPosition = position;
+            pCenterDirection = direction;
+
+            pCenterFposition = fPosition;
+            pCenterFdirection = fDirection;
+        }
+
+        public CustomSaveData(
+            SerializableVector3 playerPosition, int playerRotation, 
+            SerializableVector3 followerPosition, int followerDirection,
+            int activeScene,
+            Pokemon[][] party, Bag playerBag, List<CustomSaveEvent> saveData,
+            string saveName)
+        {
+            PlayerPosition = playerPosition;
+            PlayerDirection = playerRotation;
+
+            FollowerPosition = followerPosition;
+            FollowerDirection = followerDirection;
+
+            ActiveScene = activeScene;
+
+            Party = party;
+            PlayerBag = playerBag;
+
+            SaveData = saveData;
+            SaveName = saveName;
+
+            TimeCreated = DateTime.Now;
+        }
+    }
+}
+
+[System.Serializable]
+public enum SaveEventType
+{
+    ITEM,
+    BATTLE,
```

Comment:
Autosaving after every trigger is a lot of cons behind it. I suggest you integrate somewhere in a code a bool, that gives user/developer option to enable it.